From c7fbcd7ec148b987c6f00c59f136306ce1eb958b Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Mon, 12 Jan 2015 11:24:07 -0800
Subject: [PATCH] guard against NULL ptr during discovery from unexpected event

When demand loading drivers during discovery, iscsiadm can receive an
unexpected netlink event, like a link up, when looking for a discovery
session login status.  That could expose krecv_conn_state to a
connection without a valid recv_context pointer.
Guard against that to prevent the NULL dereference.

Signed-off-by: Chris Leech <cleech@redhat.com>
---
 usr/netlink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/usr/netlink.c b/usr/netlink.c
index 328f21c..3984727 100644
--- a/usr/netlink.c
+++ b/usr/netlink.c
@@ -1038,6 +1038,10 @@ static int krecv_conn_state(struct iscsi_conn *conn, uint32_t *state)
 		/* fatal handling error or conn error */
 		goto exit;
 
+        /* unexpected event without a receive context */
+        if (!conn->recv_context)
+                return -EAGAIN;
+
 	*state = *(enum iscsi_conn_state *)conn->recv_context->data;
 
 	ipc_ev_clbk->put_ev_context(conn->recv_context);
-- 
2.5.0

